hcp-ui-templates/ecs-existing-vpc/main.tf

```diff
@@ -0,0 +1,121 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.43"
+    }
+    hcp = {
+      source  = "hashicorp/hcp"
+      version = ">= 0.18.0"
+    }
+  }
+
+  provider_meta "hcp" {
+    module_name = "hcp-consul"
+  }
+}
+
+provider "consul" {
+  address    = hcp_consul_cluster.main.consul_public_endpoint_url
+  datacenter = hcp_consul_cluster.main.datacenter
+  token      = hcp_consul_cluster_root_token.token.secret_id
+}
+
+locals {
+  vpc_region      = "{{ .VPCRegion }}"
+  hvn_region      = "{{ .HVNRegion }}"
+  cluster_id      = "{{ .ClusterID }}"
+  vpc_id          = "{{ .VPCID }}"
+  route_table_id  = "{{ .RouteTableID }}"
+  public_subnet1  = "{{ .PublicSubnet1 }}"
+  public_subnet2  = "{{ .PublicSubnet2 }}"
+  private_subnet1 = "{{ .PrivateSubnet1 }}"
+  private_subnet2 = "{{ .PrivateSubnet2 }}"
+}
+
+provider "aws" {
+  region = local.vpc_region
+}
+
+data "aws_availability_zones" "available" {}
+
+resource "hcp_hvn" "main" {
+  hvn_id         = "${local.cluster_id}-hvn"
+  cloud_provider = "aws"
+  region         = local.hvn_region
+  cidr_block     = "172.25.32.0/20"
+}
+
+module "aws_hcp_consul" {
+  source = "hashicorp/hcp-consul/aws"
+
+  hvn             = hcp_hvn.main
+  vpc_id          = local.vpc_id
+  subnet_ids      = [local.public_subnet1, local.public_subnet2]
+  route_table_ids = [local.route_table_id]
+}
+
+resource "hcp_consul_cluster" "main" {
+  cluster_id      = local.cluster_id
+  hvn_id          = hcp_hvn.main.hvn_id
+  public_endpoint = true
+  tier            = "development"
+}
+
+resource "consul_config_entry" "service_intentions" {
+  name = "*"
+  kind = "service-intentions"
+
+  config_json = jsonencode({
+    Sources = [
+      {
+        Action     = "allow"
+        Name       = "*"
+        Precedence = 9
+        Type       = "consul"
+      },
+    ]
+  })
+}
+
+resource "hcp_consul_cluster_root_token" "token" {
+  cluster_id = hcp_consul_cluster.main.id
+}
+
+module "aws_ecs_cluster" {
+  source = "hashicorp/hcp-consul/aws//modules/hcp-ecs-client"
```

Comment:
For the other templates, I've explicitly set the `version`, i.e. `version = "0.3.0"`. My thinking there was to make sure any breaking changes we make to the module don't impact the existing terraform that the UI returns.

Do you think that is valuable to do, or should we just always pull the latest version, like here? Pulling the latest version does allow us to fix bugs easier.

Reply:
Good point. Asking the team for their recommendation.